x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/elasticsearch_assets/enrich_policy.ts

```diff
@@ -21,69 +23,79 @@ export const getFieldRetentionEnrichPolicyName = ({
   return `entity_store_field_retention_${entityType}_${namespace}_v${version}`;
 };
 
-const getFieldRetentionEnrichPolicy = (
-  unitedDefinition: UnitedEntityDefinition
-): EnrichPutPolicyRequest => {
-  const { namespace, entityType, fieldRetentionDefinition } = unitedDefinition;
-  return {
-    name: getFieldRetentionEnrichPolicyName(unitedDefinition),
-    match: {
-      indices: getEntitiesIndexName(entityType, namespace),
-      match_field: fieldRetentionDefinition.matchField,
-      enrich_fields: fieldRetentionDefinition.fields.map(({ field }) => field),
-    },
-  };
-};
-
 export const createFieldRetentionEnrichPolicy = async ({
   esClient,
-  unitedDefinition,
+  description,
+  options,
 }: {
   esClient: ElasticsearchClient;
-  unitedDefinition: UnitedEntityDefinition;
+  description: EntityEngineInstallationDescriptor;
+  options: { namespace: string };
 }) => {
-  const policy = getFieldRetentionEnrichPolicy(unitedDefinition);
-  return esClient.enrich.putPolicy(policy);
+  return esClient.enrich.putPolicy({
+    name: getFieldRetentionEnrichPolicyName({
+      namespace: options.namespace,
+      entityType: description.entityType,
+      version: description.version,
+    }),
+    match: {
+      indices: getEntitiesIndexName(description.entityType, options.namespace),
+      match_field: description.identityFields[0], // TODO figure out what happens when there are multiple identity fields
```

Comment:
do we support multiple? we should make it explicit through typing or validation if we do or don't

Reply:
We don't support multiple, and unfortunately I don't know a way that we could due to enrich policies only accepting a single `match_field`.

I've updated the "Entity Descriptions" to only allow for a single identity field, even though Entity Manager/transforms allow for multiple.

Reply:
[Resolved](https://github.com/elastic/kibana/pull/202888/commits/a75679f2a699c8bf580411f48b1573f50b463972)